c7n/resources/iam.py

```diff
@@ -1504,6 +1505,43 @@ def process(self, resources, event=None):
         return results
 
 
+@InstanceProfile.action_registry.register('add-role')
+class InstanceProfileAddRole(BaseAction):
+    """Adds specified role to IAM instance profiles that do not have a role
+
+    :example:
+
+    .. code-block:: yaml
+
+        policies:
+          - name: iam-instance-profile-add-role
+            resource: iam-profile
+            actions:
+                - type: add-role
+                  value: my-test-role
+    """
+
+    schema = type_schema('add-role',
```

Comment:
let's make this 'set-role', where a null value for the role removes the role

Reply:
@thisisshi keeping in mind this action is only meant to adds roles when instance profile does not have one, not replace or remove roles.... original thought was not to delete with this action. 

are you proposing it should work like this to delete roles?

```
actions:
      - type: set-role
```

or this?

```
actions:
      - type: set-role
        value: None
```

This doesn't appear intuitive to me, seems it would make sense to have a separate specific "remove-role" action instead?

CC: @darrendao

Reply:
I see, sounds like we'll want to add detach role permissions and support role replacement as well

Reply:
exactly, we've tried to create actions that can perform the lifecycle of the action not just a single part of the lifecycle

Reply:
@thisisshi makes sense, I've made some updates to address